### ISSUE TITLE
Hard to understand the issue on rule S2853

### ISSUE THREAD

**Issue by stas-vilchik** [SonarSource/SonarJS#746]
I'm using SonarLint plugin (version 1.1.0) for VSCode, and I'm getting an issue on the following code:

```javascript
if (task.status === 'FAILED') {
  return this.renderWhenExportFailed();
}

if (task == null) {
//  ^^^^^^^^^^^^ Change this condition so it does not always evaluate to "false"
  return this.renderWhenNoTask();
}
```

If my understanding is correct, SonarJS makes an assumption that `task` is not null, because its property is accessed `task.status`.

The problem is that I was able to understand the reason for issue **only** because I more or less know how the rule works. I cannot imagine how a regular js developer understands the issue.

**Comment by vilchik-elena** [comment]
Relates to #688

**Comment by vilchik-elena** [comment]
This rule is rewritten to be based on ESLint, closing the issue
